### ISSUE TITLE
Global Cargo.toml should specify exact subcrate versions

### ISSUE THREAD

**Issue by Canop** [TimonPost/crossterm#186]
Right now, when we import crossterm as a dependency, the sub crate versions are kind of random. And so are the bugs.

I was surprised to have a very different behavior after having just copied a project. With the exact same code and Cargo.toml. So I ran `cargo tree`:

```
dys@tour-noire:~/dev/appOne> cargo tree
appOne v0.6.1 (/home/dys/dev/appOne)
├── crossbeam v0.7.2
│   ├── cfg-if v0.1.9
│   ├── crossbeam-channel v0.3.9
│   │   └── crossbeam-utils v0.6.6
│   │       ├── cfg-if v0.1.9 (*)
│   │       └── lazy_static v1.3.0
│   ├── crossbeam-deque v0.7.1
│   │   ├── crossbeam-epoch v0.7.2
│   │   │   ├── arrayvec v0.4.11
│   │   │   │   └── nodrop v0.1.13
│   │   │   ├── cfg-if v0.1.9 (*)
│   │   │   ├── crossbeam-utils v0.6.6 (*)
│   │   │   ├── lazy_static v1.3.0 (*)
│   │   │   ├── memoffset v0.5.1
│   │   │   │   [build-dependencies]
│   │   │   │   └── rustc_version v0.2.3
│   │   │   │       └── semver v0.9.0
│   │   │   │           └── semver-parser v0.7.0
│   │   │   └── scopeguard v1.0.0
│   │   └── crossbeam-utils v0.6.6 (*)
│   ├── crossbeam-epoch v0.7.2 (*)
│   ├── crossbeam-queue v0.1.2
│   │   └── crossbeam-utils v0.6.6 (*)
│   └── crossbeam-utils v0.6.6 (*)
├── crossterm v0.9.6
│   ├── crossterm_cursor v0.2.4
│   │   └── crossterm_utils v0.2.3
│   │       └── libc v0.2.54
│   ├── crossterm_input v0.3.6
│   │   ├── crossterm_screen v0.2.3
│   │   │   └── crossterm_utils v0.2.3 (*)
│   │   ├── crossterm_utils v0.2.3 (*)
│   │   └── libc v0.2.54 (*)
│   ├── crossterm_screen v0.2.3 (*)
│   ├── crossterm_style v0.3.3
│   │   └── crossterm_utils v0.2.3 (*)
│   ├── crossterm_terminal v0.2.4
│   │   ├── crossterm_cursor v0.2.4 (*)
│   │   ├── crossterm_utils v0.2.3 (*)
│   │   └── libc v0.2.54 (*)
│   └── crossterm_utils v0.2.3 (*)
├── lazy_static v1.3.0 (*)
└── minimad v0.4.1


dys@tour-noire:~/dev/appTwo> cargo tree
appTwo v0.1.0 (/home/dys/dev/appTwo)
├── crossbeam v0.7.2
│   ├── cfg-if v0.1.9
│   ├── crossbeam-channel v0.3.9
│   │   └── crossbeam-utils v0.6.6
│   │       ├── cfg-if v0.1.9 (*)
│   │       └── lazy_static v1.3.0
│   ├── crossbeam-deque v0.7.1
│   │   ├── crossbeam-epoch v0.7.2
│   │   │   ├── arrayvec v0.4.11
│   │   │   │   └── nodrop v0.1.13
│   │   │   ├── cfg-if v0.1.9 (*)
│   │   │   ├── crossbeam-utils v0.6.6 (*)
│   │   │   ├── lazy_static v1.3.0 (*)
│   │   │   ├── memoffset v0.5.1
│   │   │   │   [build-dependencies]
│   │   │   │   └── rustc_version v0.2.3
│   │   │   │       └── semver v0.9.0
│   │   │   │           └── semver-parser v0.7.0
│   │   │   └── scopeguard v1.0.0
│   │   └── crossbeam-utils v0.6.6 (*)
│   ├── crossbeam-epoch v0.7.2 (*)
│   ├── crossbeam-queue v0.1.2
│   │   └── crossbeam-utils v0.6.6 (*)
│   └── crossbeam-utils v0.6.6 (*)
├── crossterm v0.9.6
│   ├── crossterm_cursor v0.2.5
│   │   └── crossterm_utils v0.2.4
│   │       └── libc v0.2.60
│   ├── crossterm_input v0.3.8
│   │   ├── crossterm_screen v0.2.4
│   │   │   └── crossterm_utils v0.2.4 (*)
│   │   ├── crossterm_utils v0.2.4 (*)
│   │   └── libc v0.2.60 (*)
│   ├── crossterm_screen v0.2.4 (*)
│   ├── crossterm_style v0.3.3
│   │   └── crossterm_utils v0.2.4 (*)
│   ├── crossterm_terminal v0.2.5
│   │   ├── crossterm_cursor v0.2.5 (*)
│   │   ├── crossterm_utils v0.2.4 (*)
│   │   └── libc v0.2.60 (*)
│   └── crossterm_utils v0.2.4 (*)
├── lazy_static v1.3.0 (*)
└── minimad v0.4.1

```
Bugs can't be avoided *a priori*.

But specifying a version of crossterm should be enough to determine the version of crossterm we get.

**Comment by TimonPost** [comment]
@Canop  does this PR #185 fix this?

**Comment by Canop** [comment]
I think as a rule any version of crossterm should precise the exact version (with `=`, not `^`) of its internal dependencies, so that specifying the version of crossterm (for example `"=0.10.3"`) gives us a precise version of its components.

**Comment by TimonPost** [comment]
That sounds like a valid, point, I fixed this in #189
